### PR TITLE
Add finetuning component

### PR DESCRIPTION
**Pull request by XinyuYe-Intel**
## Description

Continue the work of https://github.com/opea-project/GenAIComps/pull/236.

## Issues

`n/a`.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds new functionality)

## Dependencies

List the newly introduced 3rd party dependency if exists.

### CODE REVIEW

File: comps/finetuning/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+```bash
+export RAY_ADDRESS="ray://${ray_head_ip}:10001"
+python finetuning/finetuning_service.py
+```
+
+# 🚀2. Consume Finetuning Service
+
+## 2.1 Create fine-tuning job
+
+Assuming a training file `alpaca_data.json` is uploaded, the following script launches a finetuning job using `meta-llama/Llama-2-7b-chat-hf` as base model:
```
Comment (kevinintel) [inline comps/finetuning/README.md:46]: please add the url

File: comps/finetuning/docker/Dockerfile.finetune [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Use the same python version with ray
+FROM python:3.10.14
+
+WORKDIR /root/opea-finetune
```
Comment (kevinintel) [inline comps/finetuning/docker/Dockerfile.finetune:4]: please create new user and don't use root

File: comps/finetuning/README.md (hunk 1 above)
Comment (XinyuYe-Intel) [reply comps/finetuning/README.md]: it's done.

File: comps/finetuning/docker/Dockerfile.finetune (hunk 2 above)
Comment (XinyuYe-Intel) [reply comps/finetuning/docker/Dockerfile.finetune]: it's done.

File: comps/finetuning/handlers.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    global ray_client
+    ray_client = JobSubmissionClient() if ray_client is None else ray_client
+    ray_client.stop_job(ray_job_id)
+
+    job = running_finetuning_jobs.get(fine_tuning_job_id)
+    job.status = "cancelled"
+    return job
+
+
+async def save_content_to_local_disk(save_path: str, content):
```
Comment (lkk12014402) [inline comps/finetuning/handlers.py:157]: https://platform.openai.com/docs/api-reference/uploads/create

openai has the format for uploading, https://platform.openai.com/docs/api-reference/uploads/create. we should follow this. And If we don't have enough time, we can refine this next time

File: comps/finetuning/handlers.py [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    fine_tuning_job_id = request.fine_tuning_job_id
+
+    job = running_finetuning_jobs.get(fine_tuning_job_id)
+    if job is None:
+        raise HTTPException(status_code=404, detail=f"Fine-tuning job '{fine_tuning_job_id}' not found!")
+    output_dir = os.path.join(JOBS_PATH, job.id)
+    checkpoints = []
+    if os.path.exists(output_dir):
+        checkpoints = os.listdir(output_dir)
+    return checkpoints
```
[the remark this answers is not in the document]
Comment (XinyuYe-Intel) [reply comps/finetuning/handlers.py]: Yes, we can refine both next time.

**Review by XuhuiRen — APPROVED** [review]

**Comment by chensuyue** [comment]
Please add e2e test in another PR.

**Pull request merged**